rest/src/server/bootstrapper.js

```diff
@@ -93,23 +94,59 @@ module.exports = {
 
 	/**
 	 * Creates a REST api server.
-	 * @param {array} crossDomainConfig Configuration related to access control, contains allowed host and HTTP methods.
+	 * @param {array} config Application configuration
 	 * @param {object} formatters Formatters to use for formatting responses.
 	 * @param {object} throttlingConfig Throttling configuration parameters, if not provided throttling won't be enabled.
 	 * @returns {object} Server.
 	 */
-	createServer: (crossDomainConfig, formatters, throttlingConfig) => {
-		// create the server using a custom formatter
-		const server = restify.createServer({
+	createServer: (config, formatters, throttlingConfig) => {
+		let serverOptions = {
 			name: '', // disable server header in response
 			formatters: {
 				'application/json': formatters.json
 			}
-		});
+		};
+		if (!config.protocol)
+			winston.warn('Protocol is not configured explicitly in the configuration, defaulting to HTTPS.');
+
+		const protocol = config.protocol ? config.protocol : 'HTTPS';
```

Comment:
fyi, since this is JS, this can be written more concisely as `config.protocol || 'HTTPS'`